### PR TITLE
Create pull_request_template.md

### PR DESCRIPTION
**Pull request by MarcoPolo**
From the discussion here: https://github.com/multiformats/multiaddr/pull/138#issuecomment-1244753387

Codifies two rules we may want to follow:
* Allow 24 hours for community input before merging
* If this is a new protocol addition, don't forget to update multicodec as well.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## Summary
+<!-- What's the change? -->
+
+## Before Merge
+<!-- Anything that's needed before we merge this? -->
+- [ ] Allow 24 hours for community input
```
Comment (mriise) [inline .github/pull_request_template.md:6]:
```suggestion
- [ ] Allow at least 24 hours for community input
```

**Review by mxinden — APPROVED** [review]
:rocket: Thank you for making this easier for everyone in the future!

File: .github/pull_request_template.md (hunk 1 above)
Comment (marten-seemann) [reply .github/pull_request_template.md:6]: I agree with @mriise.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+- [ ] If this is a new protocol, has the change been applied to https://github.com/multiformats/multicodec as well?
```
Comment (marten-seemann) [inline .github/pull_request_template.md:7]: Should we ask people to link the respective PR?

**Review by marten-seemann — APPROVED** [review]

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+## Summary
+<!-- What's the change? -->
+
+## Before Merge
+<!-- Anything that's needed before we merge this? -->
+- [ ] Allow at least 24 hours for community input
+- [ ] If this is a new protocol, has the change been applied to https://github.com/multiformats/multicodec as well?
```
Comment (MarcoPolo) [inline .github/pull_request_template.md:7]:
```suggestion
- [ ] If this is a new protocol, has the change been applied to https://github.com/multiformats/multicodec as well? If so please link the multicodec PR.
```

**Pull request merged**